### PR TITLE
feat(filter): added `--exact` for exact matches

### LINKED ISSUE #50: Option to filter by whole words (not fuzzy)
For instance

echo "The quick brown fox" | gum filter

Will show up if I type `hioo` in the filter.

I'd prefer an option where it will only show up `quick` or `k bro`

### PR DESCRIPTION
**Pull request by enrichman**
Fixes #50 

### Changes
Added the `--exact` flag to perform an exact filter (not fuzzy).

To highlight the correct letters I had to override the ones matched originally, and also I had to sort the slice again to avoid the "shuffling" of some elements that were ordered with the fuzzy ranking algorithm.

### CODE REVIEW

**Comment by Justinzobel** [comment]
Ooh nice work thanks @enrichman!

File: filter/filter.go [hunk 1]
```diff
@@ -180,6 +182,11 @@ func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 			// https://github.com/sahilm/fuzzy
 			m.matches = fuzzy.Find(m.textinput.Value(), m.choices)
 
+			// For exact search return only the exact matches
+			if m.exact {
+				m.matches = exactMatches(m.textinput.Value(), m.matches)
+			}
```
Comment (dhruvmanila) [inline filter/filter.go:188]: I think we can put the fuzzy finding in the else block here as it's either exact or fuzzy.

Comment (enrichman) [reply filter/filter.go]: Ah, yes, right! We can filter directly the `choices`. And since they are stable I've also removed the sort. 👍

**Comment by maaslalani** [comment]
@enrichman would you be okay if the flag was `--fuzzy` and `--no-fuzzy`

**Comment by maaslalani** [comment]
This is how it looks, if someone uses both flags than it defaults to `--no-fuzzy`.

image

**Comment by enrichman** [comment]
Ah, that's pretty neat! Thanks for sharing.

**Comment by maaslalani** [comment]
Fantastic work on this PR, this was a heavily requested feature! Thank you so much. Will merge after CI is green 😄

**Pull request merged**